pkg/transformer/tlp_test.go

```diff
@@ -136,6 +136,51 @@ var (
 					"UNION SELECT * FROM t WHERE t.c IS NULL",
 			},
 		},
+		{
+			// normal transform with aggregate functions
+			tp:   WHERE,
+			expr: "t.c",
+			TestCase: TestCase{
+				origin: "SELECT MAX(c), MIN(c), SUM(c) FROM t",
+				expect: "SELECT MAX(c0), MIN(c1), SUM(c2) FROM " +
+					"(" +
+					"SELECT MAX(c) as c0, MIN(c) as c1, SUM(c) as c2 FROM t WHERE t.c IS TRUE " +
+					"UNION ALL SELECT MAX(c) as c0, MIN(c) as c1, SUM(c) as c2 FROM t WHERE t.c IS FALSE " +
+					"UNION ALL SELECT MAX(c) as c0, MIN(c) as c1, SUM(c) as c2 FROM t WHERE t.c IS NULL" +
+					")" +
+					" as tmp",
+			},
+		},
+		{
+			// transform with aggregate functions, partially with alias
+			tp:   WHERE,
+			expr: "t.c",
+			TestCase: TestCase{
+				origin: "SELECT MAX(c), MIN(c), SUM(c) as sum_c FROM t",
+				expect: "SELECT MAX(c0), MIN(c1), SUM(sum_c) FROM " +
+					"(" +
+					"SELECT MAX(c) as c0, MIN(c) as c1, SUM(c) as sum_c FROM t WHERE t.c IS TRUE " +
+					"UNION ALL SELECT MAX(c) as c0, MIN(c) as c1, SUM(c) as sum_c FROM t WHERE t.c IS FALSE " +
+					"UNION ALL SELECT MAX(c) as c0, MIN(c) as c1, SUM(c) as sum_c FROM t WHERE t.c IS NULL" +
+					")" +
+					" as tmp",
+			},
+		},
+		{
+			// transform with aggregate functions, partially with normal fields
+			tp:   WHERE,
+			expr: "t.c",
+			TestCase: TestCase{
+				origin: "SELECT MAX(c), MIN(c), SUM(c), c FROM t",
+				expect: "SELECT MAX(c0), MIN(c1), SUM(c2), c FROM " +
```

Comment:
Because you alias the subquery as `tmp`, I prefer to use a qualified name(tmp.c0, tmp.c1, tmp.c2) here.